src/main/java/io/stargate/sgv2/jsonapi/util/naming/NamingRule.java

```diff
@@ -0,0 +1,14 @@
+package io.stargate.sgv2.jsonapi.util.naming;
+
+/** Define a naming rule that can be applied to a string */
+public interface NamingRule {
```

Comment:
consider making this abstract class. 

you are using the name in API error messages, and so hidding a constant string for the name of the schema types. 

Replace name with schemaType using SchemaObjectType , and then add an apiName property to the enum. Use that string in the CommandResolver to build the error